azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/NetworkInterfaces.java

```diff
@@ -23,4 +24,22 @@
         SupportsDeletingById,
         SupportsDeletingByGroup,
         SupportsBatchCreation<NetworkInterface> {
+    /**
+     * List the network interfaces associated with a virtual machine scale set.
+     *
+     * @param resourceGroupName virtual machine scale set resource group name
+     * @param scaleSetName scale set name
+     * @return list of network interfaces
+     */
+    PagedList<VirtualMachineScaleSetNetworkInterface> listByVirtualMachineScaleSet(String resourceGroupName, String scaleSetName);
+
+    /**
+     * List the network interfaces associated with a specific virtual machine instance in a scale set.
+     *
+     * @param resourceGroupName virtual machine scale set resource group name
+     * @param scaleSetName scale set name
+     * @param instanceId the virtual machine scale set vm instance id
+     * @return list of network interfaces
+     */
+    PagedList<VirtualMachineScaleSetNetworkInterface> listByVirtualMachineScaleSetVMInstanceId(String resourceGroupName, String scaleSetName, String instanceId);
```

Comment:
there is a naming convention issue here. I know the Swagger spec uses this weird naming style, but it just looks wrong - VirtualMachine is spelled out in one place, but abbreviated in another in the same name.
In a way, I wish "VirtualMachine" was not part of the name, and ScaleSet should be enough, as there are no other types of scale sets. It might be too late for that kind of simplification though...
On the other hand, repeating the word "VirtualMachine" in the same name looks horrible as well.
Maybe a reasonable compromise could be: `listByVirtualMachineScaleSetInstanceId()`?